### PR TITLE
Added cert upload / delete

### PR DESCRIPTION
**Pull request by BadWolf42**
Preview :
image

```
FIXME: sometimes after an upload failed,
	it's no longer possible to retry an upload,
	modale has to be closed and reopened
	(file may stay stuck in input form field).
TODO: add documentation and maybe a tooltips

Also, removed some trailing whitespaces.
```

### CODE REVIEW

**Comment by BadWolf42** [comment]
Added public function isCertUsed($certname) in jMQTT Class
Added a ajax function to list files in certs/templates dir (currently commented)
Fixed "no longer possible to retry an upload" issue
Improvement: Check if Cert is used before delete
Improvement: Add/delete updates fields without reloading page/modale
Added Cert Add button on Broker page and blink on add:
image

File: core/class/jMQTT.class.php [hunk 1]
```diff
@@ -1529,7 +1529,41 @@ public function getBroker() {
...
+     * @param string $certname Certificate file name
+     * @return boolean true if certificat is used
+     */
+    public function isCertUsed($certname) {
+        if ($certname == $this->getConf(self::CONF_KEY_MQTT_TLS_CA)) {
+            if (boolval($this->getConf(self::CONF_KEY_MQTT_TLS))) {
+                return true;
+            } else {
+                $this->setConfiguration(self::CONF_KEY_MQTT_TLS_CA, $this->getDefaultConfiguration(self::CONF_KEY_MQTT_TLS_CA));
+                $this->save();
```
Comment (Domochip) [inline core/class/jMQTT.class.php]: this kind of clean needs to be done in preSave

File: core/class/jMQTT.class.php [hunk 2]
```diff
@@ -1529,7 +1529,41 @@ public function getBroker() {
...
+            }
+        }
+        if ($certname == $this->getConf(self::CONF_KEY_MQTT_TLS_CLI_CERT)) {
+            if (boolval($this->getConf(self::CONF_KEY_MQTT_TLS))) {
```
Comment (Domochip) [inline core/class/jMQTT.class.php]: And so this condition can be moved out around 3 if

File: desktop/php/jMQTT_broker.php [hunk 3]
```diff
@@ -183,7 +173,7 @@
 ?>
                             <div id="dTlsCaFile" class="form-group">
                                 <label class="col-lg-4 control-label">{{Autorité Personnalisée}} <sup><i class="fa fa-question-circle tooltips"
-                                title="Selectionne l'autorité de certification attendue pour le Broker.<br/>Les certificats doivent être dans : <?php echo $dir; ?>."></i></sup></label>
+                                title="Sélectionne l'autorité de certification attendue pour le Broker.<br/>Les certificats peuvent être envoyés sur Jeedom avec le bouton vert ci-dessus.<br/>Il est possible de supprimer des Certificats depuis la page de configuration générale de Plugin."></i></sup></label>
```
Comment (Domochip) [inline desktop/php/jMQTT_broker.php]: "générale de Plugin" => "générale du Plugin"

Comment (BadWolf42) [reply desktop/php/jMQTT_broker.php]: Correction done

File: core/class/jMQTT.class.php (hunk 2 above)
Comment (BadWolf42) [reply core/class/jMQTT.class.php]: Yep, in fact this only a big boolean operation now ;)

File: core/class/jMQTT.class.php (hunk 1 above)
Comment (BadWolf42) [reply core/class/jMQTT.class.php]: Ok, got it let me known if I implemented it at the right place this time.
Initially I did it that way to be able to retain the certs on a broker with TLS disabled.

**Comment by Domochip** [comment]
Time is running out, but finally, Merge!!!
Thanks again

**Pull request merged**